curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -694,17 +715,67 @@ <DATA_TYPE> void processBackgroundOperation(OperationAndData<DATA_TYPE> operatio
         }
     }
 
+    private void abortOperation(OperationAndData<?> operation, Throwable e)
+    {
+        if (operation.getCallback() == null) {
+            return;
+        }
+        CuratorEvent event;
+        if (e instanceof KeeperException)
+        {
+            event = new CuratorEventImpl(this, operation.getEventType(), ((KeeperException) e).code().intValue(), ((KeeperException) e).getPath(), null, operation.getContext(), null, null, null, null, null, null);
+        }
+        else if (getState() == CuratorFrameworkState.STARTED)
+        {
+            event = new CuratorEventImpl(this, operation.getEventType(), KeeperException.Code.SYSTEMERROR.intValue(), null, null, operation.getContext(), null, null, null, null, null, null);
```

Comment:
This `SYSTEMERROR` matches `AsyncResultImpl`'s choice. I don't have any other candidates.